### PR TITLE
Implemented file export progress dialog (#111)

### PR DESCRIPTION
**Pull request by r2d2rigo**
Added feature for issue #111 - a progress dialog that displays and refreshes while extracting data from the package. It is also cancelable in case the user doesn't want to wait for it to finish.

### CODE REVIEW

File: GUI/Forms/ExtractProgressForm.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                    await ExtractFilesAsync();
+                });
+        }
+
+        protected override void OnClosing(CancelEventArgs e)
+        {
+            cancellationTokenSource.Cancel();
+        }
+
+        private async Task CalculateFilesToExtractAsync(TreeNode root)
```
Comment (xPaw) [inline GUI/Forms/ExtractProgressForm.cs]: This is async but doesn't actually do any async work?

**Comment by xPaw** [comment]
Thanks, this looks great. One thing though, either the progress window should be closed on end, or the text should be changed to say its finished.

File: GUI/Forms/ExtractProgressForm.cs (hunk 1 above)
Comment (r2d2rigo) [reply GUI/Forms/ExtractProgressForm.cs]: Sorry, the habit of always making everything async :)

**Comment by r2d2rigo** [comment]
Done!

**Pull request merged**